tests/run_tests/driver_mag_test.py

```diff
@@ -0,0 +1,246 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from os import path
+
+import numpy as np
+from discretize.utils import active_from_xyz, mesh_builder_xyz, refine_tree_xyz
+from geoh5py.objects import Points, Surface
+from geoh5py.workspace import Workspace
+from scipy.spatial import Delaunay
+from SimPEG import utils
+
+from geoapps.utils import get_inversion_output, treemesh_2_octree
+
+
+def setup_workspace(work_dir, phys_prop):
+    project = path.join(work_dir, "mag_test.geoh5")
+    workspace = Workspace(project)
+
+    # Topography
+    xx, yy = np.meshgrid(np.linspace(-200.0, 200.0, 50), np.linspace(-200.0, 200.0, 50))
+    b = 100
+    A = 50
+    zz = A * np.exp(-0.5 * ((xx / b) ** 2.0 + (yy / b) ** 2.0))
+    topo = np.c_[utils.mkvc(xx), utils.mkvc(yy), utils.mkvc(zz)]
+    triang = Delaunay(topo[:, :2])
+
+    Surface.create(workspace, vertices=topo, cells=triang.simplices, name="topography")
+
+    # Observation points
+    xr = np.linspace(-100.0, 100.0, 20)
+    yr = np.linspace(-100.0, 100.0, 20)
+    X, Y = np.meshgrid(xr, yr)
+    Z = A * np.exp(-0.5 * ((X / b) ** 2.0 + (Y / b) ** 2.0)) + 5.0
+    points = Points.create(
+        workspace,
+        vertices=np.c_[utils.mkvc(X.T), utils.mkvc(Y.T), utils.mkvc(Z.T)],
+        name="survey",
+    )
+
+    # Create a mesh
+    h = [5.0, 5.0, 5.0]
+    padDist = np.ones((3, 2)) * 100
+    nCpad = [2, 4, 2]
+    mesh = mesh_builder_xyz(
+        points.vertices,
+        h,
+        padding_distance=padDist,
+        mesh_type="TREE",
+    )
+    mesh = refine_tree_xyz(
+        mesh,
+        topo,
+        method="surface",
+        octree_levels=nCpad,
+        octree_levels_padding=nCpad,
+        finalize=True,
+    )
+    octree = treemesh_2_octree(workspace, mesh, name="mesh")
+    active = active_from_xyz(mesh, topo, grid_reference="N")
+
+    # Model
+    model = utils.model_builder.addBlock(
+        mesh.gridCC,
+        np.zeros(mesh.nC),
+        np.r_[-20, -20, -15],
+        np.r_[20, 20, 20],
+        phys_prop,
+    )
+    model[~active] = np.nan
+    octree.add_data({"model": {"values": model[mesh._ubc_order]}})
+    octree.copy()  # Keep a copy around for ref
+
+    return workspace
+
+
+def test_susceptibility_run(tmp_path):
+    from geoapps.drivers.magnetic_scalar_inversion import MagneticScalarDriver
+    from geoapps.io.MagneticScalar.params import MagneticScalarParams
+
+    # Inducing field
+    H0 = (50000.0, 90.0, 0.0)
+    # Values stored from pre-runs
+    target = {
+        "data_norm": 716.999,
+        "phi_m": 0.006225,
+        "phi_d": 92500.0,
+    }
+    # Run the forward
+    workspace = setup_workspace(tmp_path, 0.05)
+    model = workspace.get_entity("model")[0]
+    params = MagneticScalarParams(
+        forward=True,
```

Comment:
Is this going to change since you updated the way we handle writing forward only drivers??

Reply:
we'll handle during merge with develop after 290 is in.